src/routes/transactions/helpers/timezone.helper.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * Converts a given date to a specified timezone.
+ *
+ * @param {Date} date The date object to be converted.
+ * @param {string} timeZone The target timezone (e.g., "Europe/Berlin").
+ *
+ * @returns {Date | undefined} A new Date object representing the date converted to the specified timezone. If an error occurs returns undefined
+ */
+export const convertToTimezone = (
+  date: Date,
+  timeZone: string,
+): Date | undefined => {
+  try {
+    const convertedDate = new Intl.DateTimeFormat('en-US', {
+      timeZone,
+      year: 'numeric',
+      month: '2-digit',
+      day: '2-digit',
+    }).format(date);
+
+    return new Date(convertedDate);
+  } catch {
+    return undefined;
+  }
```

Comment:
I'm apprehensive to swallow the error without feedback. In situ, we are validating the timezone so this shouldn't throw but in the future it may elsewhere so I'd sooner say we not catch it.

Reply:
I agree that it should be handled properly, but I wouldn't swallow it for now. I'd favour catching the error where it's called and log it accordingly, for example.

If we favour doing so later, then adding a comment would otherwise be beneficial.

Reply:
I completely agree that we should avoid hiding the error. Let's discuss error handling and determine the best approach to address this issue.

Reply:
Since the timezone is being validated in the controller, it should not throw an error. For now, the catch block has been removed to allow the method to pass through the general error handler if an issue occurs. ✅ 
https://github.com/safe-global/safe-client-gateway/pull/1911/commits/320e539ce28f1b73cde8ab12dec591f6b6ba5946